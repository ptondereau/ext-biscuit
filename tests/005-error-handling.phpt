--TEST--
Biscuit error handling test
--SKIPIF--
<?php
if (!extension_loaded('biscuit')) die('skip');
?>
--FILE--
<?php
// Test error handling functions
var_dump(biscuit_error_message() === "");
var_dump(biscuit_error_kind() === BISCUIT_ERROR_NONE);
var_dump(biscuit_error_check_count() === 0);

// Test invalid key pair deserialization
try {
    BiscuitKeyPair::deserialize("invalid data");
    var_dump(false); // Should not reach here
} catch (Exception $e) {
    var_dump(true);
    var_dump($e->getMessage() === "Invalid key data length");
}

// Test invalid token deserialization
try {
    BiscuitToken::deserialize("invalid token data");
    var_dump(false); // Should not reach here
} catch (Exception $e) {
    var_dump(true);
}

// Test invalid fact syntax
$builder = new BiscuitBuilder();
var_dump($builder->addFact("invalid fact") === false);

// Test invalid rule syntax
var_dump($builder->addRule("invalid rule") === false);

// Test invalid check syntax
var_dump($builder->addCheck("invalid check") === false);

// Test building token without key pair
try {
    $builder->build(null);
    var_dump(false); // Should not reach here
} catch (TypeError $e) {
    var_dump(true);
}

// Test authorizer with invalid policy
$kp = new BiscuitKeyPair();
$token = $builder->build($kp);
$auth = $token->authorizer();
var_dump($auth->addPolicy("invalid policy") === false);
?>
--EXPECT--
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true) 