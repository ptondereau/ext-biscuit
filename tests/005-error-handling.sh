#!/bin/sh

export TEST_PHP_SRCDIR='/home/ptondereau/Code/ext-biscuit'
export CC='cc'
export TEST_PHP_EXECUTABLE='/usr/bin/php8.3'
export SHELL='/usr/bin/zsh'
export LSCOLORS='Gxfxcxdxbxegedabagacad'
export SESSION_MANAGER='local/pop-os:@/tmp/.ICE-unix/3827,unix/pop-os:/tmp/.ICE-unix/3827'
export QT_ACCESSIBILITY='1'
export XDG_CONFIG_DIRS='/etc/xdg/xdg-pop:/etc/xdg'
export SSH_AGENT_LAUNCHER='gnome-keyring'
export LESS='-R'
export XDG_MENU_PREFIX='gnome-'
export GNOME_DESKTOP_SESSION_ID='this-is-deprecated'
export GTK_IM_MODULE='ibus'
export TERMINAL_EMULATOR='JetBrains-JediTerm'
export MAKE_TERMOUT='/dev/pts/0'
export GNOME_SHELL_SESSION_MODE='pop'
export SSH_AUTH_SOCK='deleted'
export GOPRIVATE='go.platform.sh'
export TERM_SESSION_ID='4f25b829-ae9e-4d03-8d5e-9e299a19c3dc'
export XMODIFIERS='@im=ibus'
export DESKTOP_SESSION='pop'
export GTK_MODULES='gail:atk-bridge'
export PWD='/home/ptondereau/Code/ext-biscuit'
export LOGNAME='ptondereau'
export XDG_SESSION_DESKTOP='pop'
export XDG_SESSION_TYPE='x11'
export GPG_AGENT_INFO='/run/user/1000/gnupg/S.gpg-agent:0:1'
export SYSTEMD_EXEC_PID='3827'
export XAUTHORITY='/run/user/1000/gdm/Xauthority'
export DESKTOP_STARTUP_ID='CLion-restart_TIME5946775'
export WINDOWPATH='2'
export HOME='/home/ptondereau'
export USERNAME='ptondereau'
export LANG='fr_FR.UTF-8'
export LS_COLORS='rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
export XDG_CURRENT_DESKTOP='pop:GNOME'
export STARSHIP_SHELL='zsh'
export INVOCATION_ID='a63ae83b05564e35b3781458eb13778b'
export MANAGERPID='3614'
export STARSHIP_SESSION_KEY='2064150891251013'
export MFLAGS=''
export XDG_SESSION_CLASS='user'
export MAKEFLAGS=''
export TERM='xterm-256color'
export ZSH='/home/ptondereau/.oh-my-zsh'
export USER='ptondereau'
export MAKE_TERMERR='/dev/pts/0'
export DISPLAY=':1'
export SHLVL='2'
export PAGER='less'
export MAKELEVEL='1'
export IJ_RESTARTER_LOG='/home/ptondereau/.cache/JetBrains/CLion2024.3/log/restarter.log'
export QT_IM_MODULE='ibus'
export DESKTOP_AUTOSTART_ID='10c091d17f7463552d174265078820819400000038270016'
export XDG_RUNTIME_DIR='/run/user/1000'
export JOURNAL_STREAM='9:37893'
export XDG_DATA_DIRS='/usr/share/pop:/usr/share/gnome:/home/ptondereau/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/'
export PATH='/home/ptondereau/.cargo/bin:/home/ptondereau/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ptondereau/.local/share/JetBrains/Toolbox/scripts:/usr/local/go/bin:/home/ptondereau/go/bin:/home/ptondereau/.local/bin'
export GDMSESSION='pop'
export DBUS_SESSION_BUS_ADDRESS='unix:path=/run/user/1000/bus'
export TOOLBOX_VERSION='2.5.4.38621'
export GIO_LAUNCHED_DESKTOP_FILE_PID='4501'
export GIO_LAUNCHED_DESKTOP_FILE='/home/ptondereau/.config/autostart/jetbrains-toolbox.desktop'
export OLDPWD='/home/ptondereau/Code/ext-biscuit'
export GOPATH='/home/ptondereau/go'
export _='/usr/bin/php8.3'
export SSH_CLIENT='deleted'
export SSH_TTY='deleted'
export SSH_CONNECTION='deleted'
export TEMP='/tmp'
export TEST_PHP_EXECUTABLE_ESCAPED=''\''/usr/bin/php8.3'\'''
export TEST_PHP_CGI_EXECUTABLE=''
export TEST_PHP_CGI_EXECUTABLE_ESCAPED=''\'''\'''
export TEST_PHPDBG_EXECUTABLE=''
export TEST_PHPDBG_EXECUTABLE_ESCAPED=''\'''\'''
export REDIRECT_STATUS='1'
export QUERY_STRING=''
export PATH_TRANSLATED='/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php'
export SCRIPT_FILENAME='/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php'
export REQUEST_METHOD='GET'
export CONTENT_TYPE=''
export CONTENT_LENGTH=''
export TZ=''
export TEST_PHP_EXTRA_ARGS=' -n -c '\''/home/ptondereau/Code/ext-biscuit/tmp-php.ini'\''  -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/home/ptondereau/Code/ext-biscuit/modules/" -d "extension=biscuit.so" -d "session.auto_start=0" -d "zlib.output_compression=Off"'
export HTTP_COOKIE=''

case "$1" in
"gdb")
    gdb --args '/usr/bin/php8.3'  -n -c '/home/ptondereau/Code/ext-biscuit/tmp-php.ini'   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/home/ptondereau/Code/ext-biscuit/modules/" -d "extension=biscuit.so" -d "session.auto_start=0" -d "zlib.output_compression=Off" -f "/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php"  2>&1
    ;;
"lldb")
    lldb -- '/usr/bin/php8.3'  -n -c '/home/ptondereau/Code/ext-biscuit/tmp-php.ini'   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/home/ptondereau/Code/ext-biscuit/modules/" -d "extension=biscuit.so" -d "session.auto_start=0" -d "zlib.output_compression=Off" -f "/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php"  2>&1
    ;;
"valgrind")
    USE_ZEND_ALLOC=0 valgrind $2 '/usr/bin/php8.3'  -n -c '/home/ptondereau/Code/ext-biscuit/tmp-php.ini'   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/home/ptondereau/Code/ext-biscuit/modules/" -d "extension=biscuit.so" -d "session.auto_start=0" -d "zlib.output_compression=Off" -f "/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php"  2>&1
    ;;
"rr")
    rr record $2 '/usr/bin/php8.3'  -n -c '/home/ptondereau/Code/ext-biscuit/tmp-php.ini'   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/home/ptondereau/Code/ext-biscuit/modules/" -d "extension=biscuit.so" -d "session.auto_start=0" -d "zlib.output_compression=Off" -f "/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php"  2>&1
    ;;
*)
    '/usr/bin/php8.3'  -n -c '/home/ptondereau/Code/ext-biscuit/tmp-php.ini'   -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -d "extension_dir=/home/ptondereau/Code/ext-biscuit/modules/" -d "extension=biscuit.so" -d "session.auto_start=0" -d "zlib.output_compression=Off" -f "/home/ptondereau/Code/ext-biscuit/tests/005-error-handling.php"  2>&1
    ;;
esac