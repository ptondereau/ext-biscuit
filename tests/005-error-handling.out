Segmentation fault (core dumped)

Termsig=11